agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -376,3 +456,168 @@ func (cs *CredentialSpecResource) MarshallPlatformSpecificFields(credentialSpecR
 func (cs *CredentialSpecResource) UnmarshallPlatformSpecificFields(credentialSpecResourceJSON CredentialSpecResourceJSON) {
 	return
 }
+
+// This function makes a copy of the local credential spec file to avoid modifying customer resources
+// Example Usage:
+// copyCredentialSpecLocalFile("credentialspecdomainless:file://gmsa-cred.json") - single file
+// copyCredentialSpecLocalFile("credentialspecdomainless:file://gmsa-dir\gmsa-cred.json") - Relative path with Windows file path separators
+func (cs *CredentialSpecResource) copyCredentialSpecLocalFile(credentialspec string) (string, error) {
+	relativeFilePath := strings.TrimPrefix(credentialspec, "credentialspecdomainless:file://")
+	dir, originalFileName := filepath.Split(relativeFilePath)
+
+	// Generate unique filename
+	taskId, err := getTaskIDfromARN(cs.taskARN)
+	if err != nil {
+		return "", err
+	}
+	containerName := cs.credentialSpecContainerMap[credentialspec]
+	copiedFileName := fmt.Sprintf("%s_%s_%s", taskId, containerName, originalFileName)
+
+	// Copy new file in same directory
+	// Read all the contents of the original file
+	bytes, err := os.ReadFile(filepath.Join(cs.credentialSpecResourceLocation, dir, originalFileName))
+	if err != nil {
+		return "", err
+	}
+
+	// Copy all the contents to the destination file
+	err = os.WriteFile(filepath.Join(cs.credentialSpecResourceLocation, dir, copiedFileName), bytes, 0777)
```

Comment:
Why do we want the file to have `0777` permissions when the `writeCredentialSpec` function writes a file with `0644`?

Reply:
agreed, will use the agreed upon 0644 permissions present